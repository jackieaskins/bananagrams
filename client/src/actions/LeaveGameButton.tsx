import { ExportOutlined } from '@ant-design/icons';
import { Popconfirm, Tooltip } from 'antd';
import { useHistory } from 'react-router-dom';

import FloatingButton from '../button/FloatingButton';

const LeaveGameButton = (): JSX.Element => {
  const { push } = useHistory();

  return (
    <Tooltip placement="left" title="Leave game">
      <Popconfirm
        cancelText="No"
        okText="Yes"
        onConfirm={() => push('/')}
        placement="topLeft"
        title="Are you sure you want to leave the game?"
      >
        <FloatingButton icon={<ExportOutlined />} shape="circle" />
      </Popconfirm>
    </Tooltip>
  );
};

export default LeaveGameButton;
