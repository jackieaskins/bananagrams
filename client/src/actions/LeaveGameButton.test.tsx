import { Popconfirm } from 'antd';
import { shallow } from 'enzyme';

import LeaveGameButton from './LeaveGameButton';

const mockPush = jest.fn();
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockPush,
  }),
}));

describe('<LeaveGameButton />', () => {
  const renderComponent = (propOverrides = {}) =>
    shallow(<LeaveGameButton {...propOverrides} />);

  it('renders floating button with confirmation and tooltip', () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  it('confirms redirect to homepage', () => {
    renderComponent().find(Popconfirm).props().onConfirm();

    expect(mockPush).toHaveBeenCalledWith('/');
  });
});
