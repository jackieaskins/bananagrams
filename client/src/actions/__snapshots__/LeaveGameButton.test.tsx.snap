// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeaveGameButton /> renders floating button with confirmation and tooltip 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="left"
  title="Leave game"
  transitionName="zoom-big-fast"
>
  <ForwardRef
    cancelText="No"
    disabled={false}
    icon={<ForwardRef(ExclamationCircleFilled) />}
    okText="Yes"
    okType="primary"
    onConfirm={[Function]}
    placement="topLeft"
    title="Are you sure you want to leave the game?"
    transitionName="zoom-big"
    trigger="click"
  >
    <FloatingButton
      icon={<ForwardRef(ExportOutlined) />}
      shape="circle"
    />
  </ForwardRef>
</Tooltip>
`;
